<!DOCTYPE html>
<html>

<head>
	<meta charset="utf-8">
	<meta http-equiv="X-UA-Compatible" content="IE=edge">
	<title>MSABrowser | MLH1 Example</title>
	<link type="text/css" rel="stylesheet" href="../tool.css" media="screen" />
	<script src="https://html2canvas.hertzen.com/dist/html2canvas.min.js" type="text/javascript"></script>
	<script src="https://code.jquery.com/jquery-3.3.1.min.js"
		integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
	<script src="../MSAViewer.js"></script>
</head>

<body>
	<h1>Evolutionary/Comparative Genomics Study | MLH1 Example</h1>
	<article>
		In this example, human MLH1 sequence and its homologous sequences are aligned and human sequence annotations are added. <br> 
		Additionaly, some genetic variations are also included from different sources such as gnomAD and ClinVar. 
		For example, you can view a particular <a href="javascript:void(0)" onclick="viewer.goToVar(1,33)"> modification and variation on human</a> or 
		<a href="javascript:void(0)" onclick="viewer.goToVar(2,17)"> this variation on mouse</a> or 
		by clicking the link.
	</article>

	<!-- MSAViewer Demo -->
	<section id="MSAViewerDemo"></section>
	<!-- end of MSAViewer Demo -->


	<script type="text/javascript" charset="utf-8">

		var viewer;
		var annotations = [
			{
				'annotation_id':'PF13589.6',
				'annotation_name':'HATPase_c_3',
				'annotation_external_link':'https://pfam.xfam.org/family/PF13589.6',
				'annotation_start_point':'29',
				'annotation_end_point':'131'
			},

			{
				'annotation_id':'PF01119.19',
				'annotation_name':'DNA_mis_repair',
				'annotation_external_link':'https://pfam.xfam.org/family/PF01119.19',
				'annotation_start_point':'216',
				'annotation_end_point':'334'
			},

			{
				'annotation_id':'PF16413.5',
				'annotation_name':'Mlh1_C',
				'annotation_external_link':'https://pfam.xfam.org/family/PF16413.5',
				'annotation_start_point':'502',
				'annotation_end_point':'756'
			}
		];

		var variations = [
			{ 'sequence': 1, 'position': 6, 'note': "rsNumber: rs1366752604 <br>p.Gly6Trp <br>missense_variant <br> Frequency: 3.2E-5", 'source': "gnomAD <a href='https://gnomad.broadinstitute.org/variant/3-37035054-G-T' target='_blank'> For details, visit here</a>" },
			{ 'sequence': 1, 'position': 33, 'note': "rsNumber: rs878853795 <br> single nucleotide variant <br> Lynch syndrome", 'source': "ClinVar <a href='https://www.ncbi.nlm.nih.gov/clinvar/29040541/' target='_blank'> For details, visit here</a>" },
			{ 'sequence': 1, 'position': 33, 'note': "PTM Type: Ubiquitination ", 'source': "PTM" },
			{ 'sequence': 2, 'position': 17, 'note': "A variation in the 23rd position of this mouse protein", 'source': "Surname et al., 2020" },
			{ 'sequence': 3, 'position': 2, 'note': "A special annotation for this position can be added such as <br> G->K : VUS", 'source': "Data Source" },
			{ 'sequence': 5, 'position': 82, 'note': "A special note about modifications at this position can be added for this organism!", 'source': "PTM" }
		];

		$.get("sample_files/mlh1.txt", (fasta) => {

			viewer = new MSAViewer({
				id: "MSAViewerDemo",
				msa: MSAProcessor({
					fasta: fasta,
					hasConsensus: true
				}),
				title: "Human: MLH1",
				annotations: annotations,
				variations: variations,
				colorSchema: "clustal",
				resetOnScroll: true
			});

			viewer.export("MSA_export.fasta");
			
		});

	</script>
</body>

</html>